tools/analyzer_plugin/lib/src/component_usage.dart

```diff
@@ -40,6 +42,29 @@ class FluentComponentUsage {
     return null;
   }
 
+  /// Returns the name of the `UiComponent` class instance for the usage, complete with any
+  /// [namespaceName] if one exists.
+  String get componentClassTypeName {
+    if (componentName == null) return null;
+    if (namespaceName == null) return '${componentName}Component';
+
+    return '$namespaceName.${componentName}Component';
+  }
+
+  /// The name of the imported usage namespace.
+  String get namespaceName {
+    if (builder is MethodInvocation) {
+      return (builder as MethodInvocation).target.tryCast<Identifier>()?.name;
+    } else if (builder is FunctionExpressionInvocation) {
```

Comment:
In which case would this be a `FunctionExpressionInvocation`?

Reply:
That is what it showed up as for an invoked factory I think. If you put a breakpoint in here - I think the MethodInvocation only triggers for Dom components maybe?